internal/querycoord/impl.go

```diff
@@ -1200,6 +1200,16 @@ func (qc *QueryCoord) GetShardLeaders(ctx context.Context, req *querypb.GetShard
 		shardLeaderLists = append(shardLeaderLists, shard)
 	}
 
+	// all replicas are not available
+	if len(shardLeaderLists) == 0 {
```

Comment:
Why not check this in Proxy, which makes sense for me

Reply:
Either side will do. It's just weird for me that return a list with zero element and no error.